From 537c2a244fb1174f508e6c56be8033bca30472e3 Mon Sep 17 00:00:00 2001
From: Daniel Scally <djrscally@gmail.com>
Date: Sun, 17 Jan 2021 19:08:18 +0000
Subject: [PATCH 102/103] media: i2c: Add reset pin toggling to ov5693

The ov5693 has an xshutdown pin which can be present and, if so, needs
toggling as part of power on sequence.

Add calls to handle the reset GPIO

Signed-off-by: Daniel Scally <djrscally@gmail.com>
---
 drivers/media/i2c/ov5693.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/i2c/ov5693.c b/drivers/media/i2c/ov5693.c
index 32485e4ed42b..f9ced52ad37a 100644
--- a/drivers/media/i2c/ov5693.c
+++ b/drivers/media/i2c/ov5693.c
@@ -1085,6 +1085,8 @@ static int __power_up(struct v4l2_subdev *sd)
 	if (ret)
 		goto fail_power;
 
+	gpiod_set_value_cansleep(sensor->reset, 0);
+
 	__cci_delay(up_delay);
 
 	return 0;
@@ -1103,6 +1105,8 @@ static int power_down(struct v4l2_subdev *sd)
 
 	dev->focus = OV5693_INVALID_CONFIG;
 
+	gpiod_set_value_cansleep(sensor->reset, 1);
+
 	clk_disable_unprepare(dev->clk);
 
 	if (dev->indicator_led)
-- 
2.17.1

